tools/verify_images.py

```diff
@@ -0,0 +1,117 @@
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Utility script to verify that all images have alt text"""
+
+import multiprocessing
+import sys
+import glob
+
+# Dictionary to allowlist lines of code that the checker will not error
+# Format: {"file_path": [list_of_line_numbers]}
+ALLOWLIST_MISSING_ALT_TEXT = {
+    "qiskit_ibm_runtime/fake_provider/__init__.py": [34, 55, 63],
+    "qiskit_ibm_runtime/transpiler/passes/scheduling/dynamical_decoupling.py": [56, 88],
+    "qiskit_ibm_runtime/transpiler/passes/scheduling/__init__.py": [
+        51,
+        102,
+        137,
+        152,
+        176,
+        207,
+        226,
+        240,
+        256,
+        274,
+        294,
+        313,
+        328,
+        342,
+        363,
+        382,
+        406,
+    ],
+}
+
+
+def is_image(line) -> bool:
+    return line.strip().startswith(".. image:") or line.strip().startswith(".. plot:")
+
+
+def allowlist(filename, line_num) -> bool:
+    return (
+        filename in ALLOWLIST_MISSING_ALT_TEXT and line_num in ALLOWLIST_MISSING_ALT_TEXT[filename]
+    )
+
+
+def validate_image(file_path) -> tuple[str, list[str]]:
+    """Validate all the images of a single file"""
+    invalid_images: list[str] = []
+
+    with open(file_path, encoding="utf8") as fd:
+        lines = fd.readlines()
+
+        for index, line in enumerate(lines):
+            if is_image(line) and not allowlist(file_path, index + 1):
+                options: list[str] = []
+                option_idx = index
+
+                while option_idx + 1 < len(lines):
+                    option_idx += 1
+                    option_line = lines[option_idx].strip()
+
+                    if not option_line.startswith(":"):
+                        break
+
+                    options.append(option_line)
+
+                alt_exists = any(option.startswith(":alt:") for option in options)
+                nofig_exists = any(option.startswith(":nofig:") for option in options)
+
+                # Only `.. plot::`` directives without the `:nofig:` option are required to have alt text.
+                # All `.. image::` directives need alt text and they don't have a `:nofig:` option.
+                if not alt_exists and not nofig_exists:
+                    invalid_images.append(f"- Error in line {index + 1}: {line}")
+
+    return (file_path, invalid_images)
+
+
+def main() -> None:
+    files = glob.glob("qiskit_ibm_runtime/**/*.py", recursive=True)
+
+    with multiprocessing.Pool() as pool:
+        results = pool.map(validate_image, files)
+
+    failed_files = list(filter(lambda x: len(x[1]), results))
+
+    if len(failed_files):
+        sys.stderr.write("💔 Some images are missing the alt text\n\n")
```

Comment:
You can print to stderr with `print("", file=sys.stderr)`. It will add the last `\n` for you.